src/hotspot/share/utilities/spinCriticalSection.hpp

```diff
@@ -22,23 +22,35 @@
  *
  */
 
-#ifndef SHARE_JFR_UTILITIES_JFRSPINLOCKHELPER_HPP
-#define SHARE_JFR_UTILITIES_JFRSPINLOCKHELPER_HPP
+#ifndef SHARE_UTILITIES_SPINCRITICALSECTION_HPP
+#define SHARE_UTILITIES_SPINCRITICALSECTION_HPP
 
 #include "runtime/javaThread.hpp"
 
-class JfrSpinlockHelper {
- private:
+// Ad-hoc mutual exclusion primitive: spin critical section,
+// which employs a spin lock.
+//
+// We use this critical section only for low-contention code, and
+// when it is know that the duration is short. To be used where
+// we're concerned about native mutex_t or HotSpot Mutex:: latency.
+class SpinCriticalSection {
+private:
+  // We use int type as 32-bit atomic operation is the most performant
+  // compared to  smaller/larger types.
   volatile int* const _lock;
 
- public:
-  JfrSpinlockHelper(volatile int* lock) : _lock(lock) {
-    Thread::SpinAcquire(_lock);
+  // Low-level leaf-lock primitives used to implement synchronization.
+  // Not for general synchronization use.
```

Comment:
This comment seems like it contains info that ought to be part of the class description, rather
than on an implementation detail.